src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -313,6 +328,58 @@ public async Task<AuthenticationResult> GetAuthenticationResultForUserAsync(
             }
         }
 
+        private async Task<AuthenticationResult?> TryGetAuthenticationResultForConfidentialClientUsingRopcAsync(IConfidentialClientApplication application, IEnumerable<string> scopes, ClaimsPrincipal? user, MergedOptions mergedOptions, TokenAcquisitionOptions? tokenAcquisitionOptions)
+        {
+            if (user != null && user.HasClaim(c => c.Type == ClaimConstants.Username) && user.HasClaim(c => c.Type == ClaimConstants.Password))
+            {
+                string username = user.FindFirst(ClaimConstants.Username)?.Value ?? string.Empty;
+                string password = user.FindFirst(ClaimConstants.Password)?.Value ?? string.Empty;
+
+                if (user.GetMsalAccountId() != null)
```

Comment:
It's never going to work as the user is created from the ClaimsPrincipalFactory.FromUsernamePassword(), and it does not have the home tenant id / home user id which are the components of GetMsalAccountId()

Reply:
I am adding these claims below [here](https://github.com/AzureAD/microsoft-identity-web/pull/3103/files#diff-3600735102cf2582ea4dc8277f799690b8a8135f4971e0be919fbecd053ae2caR370) after the first call executes